apps/vs-code-designer/src/app/commands/createLogicApp/createLogicAppSteps/HybridLogicAppsSteps/ContainerAppNameStep.ts

```diff
@@ -11,24 +11,23 @@ import type { ContainerAppsAPIClient } from '@azure/arm-appcontainers';
 
 export class ContainerAppNameStep extends AzureWizardPromptStep<ILogicAppWizardContext> {
   public async prompt(context: ILogicAppWizardContext): Promise<void> {
-    const hasValidName = await ContainerAppNameStep.isNameAvailable(context, context.resourceGroup.name, context.newSiteName);
+    const validationError = await ContainerAppNameStep.validateName(context, context.newSiteName);
 
-    if (hasValidName) {
+    if (!validationError) {
       return;
     }
 
     const prompt: string = localize(
       'newLogicAppName',
-      'Enter a new logic app name, the logic app "{0}" already exists in resource group "{1}".',
+      `Enter a new logic app name, ${validationError}`,
       context.newSiteName,
       context.resourceGroup.name
     );
 
     context.newSiteName = (
       await context.ui.showInputBox({
         prompt,
-        validateInput: ContainerAppNameStep.validateInput,
-        asyncValidationTask: (name: string) => this.validateNameAvailable(context, name),
+        asyncValidationTask: (name: string) => ContainerAppNameStep.validateName(context, name),
```

Comment:
Is there any specific reason on why we are not validating the name again if there is a validationError?
For example:

- User first put a name that ends up in validationError above. 
- So it enters again to ask the user for the name
- How does the user will know if the new name does or doesn't exists already in the resource group?

Reply:
oh I out the validateInput logic inside the `validateName` method, so it will actually do that

Reply:
Yep I noticed that, but for the new inputBox, do we validate it while the user is typing the new name? thats what I mean

Reply:
oh yeah I was checking that what's the difference in async and without, I see it now. I'll update

Reply:
Sure, thanks!